viewing_party/party.py

```diff
@@ -0,0 +1,184 @@
+# Wave 1
+
+def create_movie(movie_title, genre, rating):  
+  new_movie = {}
+  
+  if not movie_title or not genre or not rating:
+    return None  
+  
+  else:
+    new_movie["title"] = movie_title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    
+  return new_movie
+
+
+def add_to_watched(user_data, movie):
+
+  if not "watched" in user_data:
+    user_data["watched"] = [movie]
+    
+  else: 
+    user_data["watched"].append(movie)
+    
+  return user_data
+  
+  
+def add_to_watchlist(user_data, movie):
+  updated_dict = user_data
+  
+  if not "watchlist" in user_data:
+    user_data["watchlist"] = [movie]
+    
+  else: 
+    user_data["watchlist"].append(movie)
+    
+  return user_data
+
+
+def watch_movie(user_data, title):
+  
+  movie_found = False
+  index_to_be_removed = -1  # set to -1 bcs invalid, can't set it to other bcs those would be valid
+  
+  for index, temp_movie_dict in enumerate(user_data["watchlist"]):
+    if temp_movie_dict["title"] == title: #found movie to be removed
+      index_to_be_removed = index #the index of the movie
+      movie_found = True
+      user_data["watched"].append(temp_movie_dict)
+    
+  if movie_found:
+    user_data["watchlist"].pop(index_to_be_removed)
+  
+  return user_data
+
+
+# Wave 2
+
+def get_watched_avg_rating(user_data):
+  user_rating_total = 0.0
+  average = 0.0
+  watched_list = user_data["watched"]
+  
+  if watched_list != []:
+    for movie_dict in watched_list:
+      user_rating_total += movie_dict["rating"] 
+    average = user_rating_total / len(watched_list)
+    
+  return average
+
+      
+def get_most_watched_genre(user_data):
+  popular_genre_list = []
+  count = 0
+  watched_list = user_data["watched"]
+  popular_genre = None
+  
+  if watched_list != []:
+    for movie_dict in watched_list:
+      popular_genre_list.append(movie_dict["genre"])
+      
+    for genre_name in popular_genre_list:
+      
+      if popular_genre_list.count(genre_name) > count:
+        count = popular_genre_list.count(genre_name)
+        popular_genre = genre_name
+  
+  return popular_genre
+
+
+# Wave 3
+
+def get_unique_watched(user_data):
+    
+  # Movies the user has watched, but none of their friends have watched
+  
+  friends_list = user_data
+  user_movies_list = []
+  friends_movies_list = []
+  user_unique_movies = []
+  friends_unique_movies = {}
+  friends_movie_dict_list = user_data["friends"]
```

Comment:
There are a few variables here that aren't actually used in this function. Try not to create variables that aren't used.

Additionally, in Python it's more typical to initialize a variable just before we need to. Doing so can help protect us from referencing a variable by accident (by typo) before we mean to, since accessing an uninitialized variable causes an error.

Some languages do require that variable declarations come in a chunk before anything else, but this isn't the case for python.